lib/demon/sidekiq.rb

```diff
@@ -11,6 +13,37 @@ def self.after_fork(&blk)
     blk ? (@blk = blk) : @blk
   end
 
+  def self.format(message)
+    "[#{Time.now.utc.strftime("%Y-%m-%dT%H:%M:%S.%6N")} ##{Process.pid}] #{message}"
+  end
+
+  def self.log(message, level: :info)
+    # We use a IO pipe and log messages using the logger in a seperate thread to avoid the `log writing failed. can't be called from trap context`
+    # error message that is raised when trying to log from within a `Signal.trap` block.
+    if logger
+      if !defined?(@logger_read_pipe)
```

Comment:
I adapted this from https://gist.github.com/mvidner/bf12a0b3c662ca6a5784#the-self-pipe-trick